### PR TITLE
Feat: Implement API for Tracking Orphaned Tasks

### PR DESCRIPTION
**Pull request by VinuB-Dev**
Date: 15/11/2024
Developer Name: Vignesh B S

---

## Issue Ticket Number
Closes Real-Dev-Squad/website-dashboard#888

## Description

Implemented changes in tasks api for Tracking Orphaned Tasks

- `tasks?orphaned=true` - Retrieves tasks assigned to users who have departed the Discord server.

Added test cases for the above api.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-11-15 at 3 53 50 PM

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-11-15 at 3 54 09 PM

</details>

## Additional Notes

[PRD](https://docs.google.com/document/d/1NCjJQ6RJlQGqdZyr8r2myuaWekYIoEORLr94fKRaCvo/edit?addon_store&tab=t.0#heading=h.cggvkhhvicot)
[Design Doc](https://docs.google.com/document/d/1ZYhy21DLHtTnXfKRlDH-7EVlKrMV2PyZs1_6czTozDk/edit?addon_store&tab=t.0#heading=h.cggvkhhvicot)
Have moved the updateTaskStatusToDone function to models as it was directly interacts with the database and performing operations such as updating task documents. This was also causing circular dependency issues.

### CODE REVIEW

**Review by pankajjs — APPROVED** [review]

File: controllers/tasks.js [hunk 1]
```diff
@@ -134,7 +135,19 @@ const fetchPaginatedTasks = async (query) => {
...
+      size,
+      prev,
+      next,
+      q: queryString,
+      assignee,
+      title,
+      userFeatureFlag,
+      orphaned,
+      dev,
+    } = req.query;
```
Comment (pankajjs) [inline controllers/tasks.js:150]: Are we using all these query?

**Review by pankajjs — CHANGES_REQUESTED** [review]
Left comment

File: controllers/tasks.js (hunk 1 above)
Comment (VinuB-Dev) [reply controllers/tasks.js:150]: yeah I only added orphaned and dev, it's prettier change.

**Comment by VinuB-Dev** [comment]
Splitting this PR into smaller parts for easier review. This PR will serve as the main split: https://github.com/Real-Dev-Squad/website-backend/pull/2267. Closing this PR as part of the process.

**Pull request closed without merging**